### PR TITLE
refactor: Integrate `consistent-type-imports` in FE packages (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/N8N-6054/integrate-consistent-type-imports-in-fe-packages

e2e pass: https://github.com/n8n-io/n8n/actions/runs/4784768930

### CODE REVIEW

**Review by cstuncsik — APPROVED** [review]

File: packages/editor-ui/src/api/workflows.ts [hunk 1]
```diff
@@ -1,11 +1,6 @@
...
-import {
-	ExecutionFilters,
-	ExecutionOptions,
-	ExecutionStatus,
-	IDataObject,
-	WorkflowExecuteMode,
-} from 'n8n-workflow';
+import type { IExecutionsCurrentSummaryExtended, IRestApiContext } from '@/Interface';
+import type { ExecutionFilters, ExecutionOptions, IDataObject } from 'n8n-workflow';
+import { ExecutionStatus, WorkflowExecuteMode } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/api/workflows.ts:3]: These two should also be import type, no? But also, they are not used in this file so could be deleted

File: packages/editor-ui/src/components/Node.vue [hunk 2]
```diff
@@ -182,7 +182,8 @@ import { nodeHelpers } from '@/mixins/nodeHelpers';
 import { workflowHelpers } from '@/mixins/workflowHelpers';
 import { pinData } from '@/mixins/pinData';
 
-import { IDataObject, INodeTypeDescription, ITaskData, NodeHelpers } from 'n8n-workflow';
+import type { INodeTypeDescription, ITaskData } from 'n8n-workflow';
+import { IDataObject, NodeHelpers } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/Node.vue:186]: IDataObject is type, but it's not used in the file

File: packages/editor-ui/src/components/Node.vue [hunk 3]
```diff
@@ -191,12 +192,8 @@ import mixins from 'vue-typed-mixins';
...
 import { get } from 'lodash-es';
 import { getStyleTokenValue, getTriggerNodeServiceName } from '@/utils';
-import {
-	IExecutionsSummary,
-	INodeUi,
-	INodeUpdatePropertiesInformation,
-	XYPosition,
-} from '@/Interface';
+import type { IExecutionsSummary, INodeUi, XYPosition } from '@/Interface';
+import { INodeUpdatePropertiesInformation } from '@/Interface';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/Node.vue:196]: INodeUpdatePropertiesInformation is type, but it's not used in the file

File: packages/editor-ui/src/components/Node/NodeCreator/MainPanel.vue [hunk 4]
```diff
@@ -79,12 +79,9 @@
...
 <script setup lang="ts">
 import { reactive, toRefs, getCurrentInstance, computed, onUnmounted, ref } from 'vue';
-import {
-	INodeTypeDescription,
-	INodeActionTypeDescription,
-	INodeTypeNameVersion,
-} from 'n8n-workflow';
-import {
+import type { INodeTypeDescription, INodeActionTypeDescription } from 'n8n-workflow';
+import { INodeTypeNameVersion } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/Node/NodeCreator/MainPanel.vue:83]: INodeTypeNameVersion is type but not used in this file

File: packages/editor-ui/src/components/ParameterInput.vue [hunk 5]
```diff
@@ -341,6 +339,7 @@ import {
 	INodePropertyCollection,
 	NodeParameterValueType,
 } from 'n8n-workflow';
+import { NodeHelpers, NodeParameterValue } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/ParameterInput.vue:342]: NodeParameterValue is type but not used in the file

File: packages/editor-ui/src/components/ParameterInputWrapper.vue [hunk 6]
```diff
@@ -42,21 +42,21 @@
...
+import type {
 	INodeProperties,
 	INodePropertyMode,
-	IRunData,
-	isResourceLocatorValue,
 	NodeParameterValue,
 	NodeParameterValueType,
 } from 'n8n-workflow';
-import { INodeUi, IUpdateInformation, TargetItem } from '@/Interface';
+import { IRunData, isResourceLocatorValue } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/ParameterInputWrapper.vue:58]: IRunData is type but not used in the file

File: packages/editor-ui/src/components/ResourceLocator/ResourceLocator.vue [hunk 7]
```diff
@@ -135,25 +135,27 @@
...
 	INodeCredentials,
 	INodeListSearchItems,
-	INodeListSearchResult,
 	INodeParameterResourceLocator,
 	INodeParameters,
 	INodeProperties,
 	INodePropertyMode,
 	NodeParameterValue,
 } from 'n8n-workflow';
+import { INodeListSearchResult } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/ResourceLocator/ResourceLocator.vue:149]: INodeListSearchResult is type but not used in the file

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue [hunk 8]
```diff
@@ -180,20 +180,23 @@ import { useNDVStore } from '../../stores/ndv';
...
 import ParameterInputList from '@/components/ParameterInputList.vue';
 import NodeCredentials from '@/components/NodeCredentials.vue';
-import { IMenuItem, INodeUi, ITab, IUpdateInformation } from '../../Interface';
+import type { IMenuItem, INodeUi, ITab, IUpdateInformation } from '../../Interface';
+import type {
+	IDataObject,
+	INodeCredentials,
+	NodeParameterValue,
+	MessageEventBusDestinationOptions,
+} from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:189]: Should enums be also imported as type? If so, `MessageEventBusDestinationTypeNames` should be included

File: packages/editor-ui/src/components/Sticky.vue [hunk 9]
```diff
@@ -64,14 +64,15 @@ import { nodeBase } from '@/mixins/nodeBase';
...
+import type {
 	INodeUi,
 	INodeUpdatePropertiesInformation,
 	IUpdateInformation,
 	XYPosition,
 } from '@/Interface';
 
-import { IDataObject, INodeTypeDescription } from 'n8n-workflow';
+import type { INodeTypeDescription } from 'n8n-workflow';
+import { IDataObject } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/Sticky.vue:75]: IDataObject is type but not in the file

File: packages/editor-ui/src/components/VariableSelector.vue [hunk 10]
```diff
@@ -36,16 +36,12 @@ import {
...
 
 import VariableSelectorItem from '@/components/VariableSelectorItem.vue';
-import {
-	IExecutionResponse,
-	INodeUi,
-	IVariableItemSelected,
-	IVariableSelectorOption,
-} from '@/Interface';
+import type { INodeUi, IVariableItemSelected, IVariableSelectorOption } from '@/Interface';
+import { IExecutionResponse } from '@/Interface';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/VariableSelector.vue:44]: IExecutionResponse is type but not in the file

File: packages/editor-ui/src/mixins/restApi.ts [hunk 11]
```diff
@@ -16,11 +16,10 @@ import {
 	IWorkflowShortResponse,
 	IRestApi,
 	IWorkflowDataUpdate,
-	INodeTranslationHeaders,
 } from '@/Interface';
+import { INodeTranslationHeaders } from '@/Interface';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/mixins/restApi.ts:20]: INodeTranslationHeaders is type but not included in the file

File: packages/editor-ui/src/mixins/workflowHelpers.ts [hunk 12]
```diff
@@ -19,20 +19,17 @@ import {
...
 	IWorkflowIssues,
 	IWorkflowDataProxyAdditionalKeys,
 	Workflow,
-	NodeHelpers,
 	IExecuteData,
 	INodeConnection,
 	IWebhookDescription,
-	deepCopy,
 } from 'n8n-workflow';
+import { INodeTypeData, IPinData, NodeHelpers, deepCopy } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/mixins/workflowHelpers.ts:30]: INodeTypeData, IPinData are types but not in the file

File: packages/editor-ui/src/permissions.ts [hunk 13]
```diff
@@ -4,13 +4,8 @@
...
 
-import {
-	IUser,
-	ICredentialsResponse,
-	IRootState,
-	IWorkflowDb,
-	EnvironmentVariable,
-} from '@/Interface';
+import type { IUser, ICredentialsResponse, IWorkflowDb } from '@/Interface';
+import { IRootState, EnvironmentVariable } from '@/Interface';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/permissions.ts:8]: None of these are used in the file, the import could be removed

File: packages/editor-ui/src/stores/segment.ts [hunk 14]
```diff
@@ -6,10 +6,10 @@ import {
 	SET_NODE_TYPE,
 	WEBHOOK_NODE_TYPE,
 } from '@/constants';
-import { ITelemetryTrackProperties } from 'n8n-workflow';
+import type { ITelemetryTrackProperties } from 'n8n-workflow';
 import { defineStore } from 'pinia';
 import { useSettingsStore } from '@/stores/settings';
-import { INodeTypeDescription, IRun } from 'n8n-workflow';
+import type { INodeTypeDescription, IRun } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/stores/segment.ts:12]: The two `n8n-workflow` imports could be merged

File: packages/editor-ui/src/stores/workflows.ts [hunk 15]
```diff
@@ -54,9 +52,9 @@ import {
 	IRunExecutionData,
 	ITaskData,
 	IWorkflowSettings,
-	NodeHelpers,
-	Workflow,
 } from 'n8n-workflow';
+import type { IDataObject } from 'n8n-workflow';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/stores/workflows.ts:56]: Could be merged with import above

File: packages/editor-ui/src/utils/typeGuards.ts [hunk 16]
```diff
@@ -1,6 +1,6 @@
-import { NewCredentialsModal } from './../Interface';
-import { INodeParameterResourceLocator } from 'n8n-workflow';
-import { ICredentialsResponse } from '@/Interface';
+import type { NewCredentialsModal } from './../Interface';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/utils/typeGuards.ts:1]: Is same as `@/Interface` import below

File: packages/editor-ui/src/utils/typesUtils.ts [hunk 17]
```diff
@@ -1,6 +1,7 @@
 import dateformat from 'dateformat';
-import { IDataObject, jsonParse } from 'n8n-workflow';
-import { Schema, Optional, Primitives } from '@/Interface';
+import type { IDataObject } from 'n8n-workflow';
+import { jsonParse } from 'n8n-workflow';
+import type { Schema, Optional, Primitives } from '@/Interface';
 import { isObj } from '@/utils/typeGuards';
 import { generatePath } from '@/utils/mappingUtils';
 import { DateTime } from 'luxon';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/utils/typesUtils.ts:7]: DateTime is a type but not used in this file

File: packages/editor-ui/src/utils/userUtils.ts [hunk 18]
```diff
@@ -69,7 +69,7 @@ import {
 	IPersonalizationSurveyVersions,
 	IUser,
 } from '@/Interface';
-import { ILogInStatus, IRole, IUserPermissions } from '@/Interface';
+import type { ILogInStatus, IRole, IUserPermissions } from '@/Interface';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/utils/userUtils.ts:72]: Could be merged with the import above

File: packages/editor-ui/src/views/CredentialsView.vue [hunk 19]
```diff
@@ -45,21 +45,22 @@
 
 <script lang="ts">
 import { showMessage } from '@/mixins/showMessage';
-import { ICredentialsResponse, ICredentialTypeMap, IUser } from '@/Interface';
+import type { ICredentialsResponse, ICredentialTypeMap } from '@/Interface';
+import { IUser } from '@/Interface';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/views/CredentialsView.vue:49]: This is a type but not used in the file

**Review by OlegIvaniv — CHANGES_REQUESTED** [review]
Nice one! I found a few cases where we could adjust/remove imports. It seems like the linting is not triggered for imports which are not used

File: packages/editor-ui/src/api/workflows.ts (hunk 1 above)
Comment (OlegIvaniv) [inline packages/editor-ui/src/api/workflows.ts:3]: These are types but not used in the file

**Comment by janober** [comment]
Got released with [`n8n@0.226.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.226.1)

**Pull request merged**